src/session/session_helper.c

```diff
@@ -159,26 +156,28 @@ __wt_session_set_last_error(
     if (!F_ISSET(session, WT_SESSION_SAVE_ERRORS))
         return (0);
 
-    /* Format the error message string. */
-    WT_RET(__wt_scr_alloc(session, 0, &buf));
-    WT_VA_ARGS_BUF_FORMAT(session, buf, fmt, false);
-    err_msg_content = buf->data;
-
-    /* Only set the error if it results in a change. */
-    if (err_info->err == err && err_info->sub_level_err == sub_level_err &&
-      err_info->err_msg != NULL && strcmp(err_info->err_msg, err_msg_content) == 0)
-        goto err;
-
-    /* Free the last error message string. */
-    __wt_free(session, err_info->err_msg);
-
-    /* Load error codes and message content into err_info. */
-    WT_ERR(__wt_calloc(session, buf->size + 1, 1, &(err_info->err_msg)));
-    WT_ERR(__wt_snprintf(err_info->err_msg, buf->size + 1, "%s", err_msg_content));
+    /*
+     * Load error codes and message into err_info. If the message is empty or is NULL (indicating
```

Comment:
Do we need plural here?
```suggestion
     * Load the error code and message into err_info. If the message is empty or is NULL (indicating
```

Reply:
Should be short for error code + sub level error code - I can make that more explicit if necessary

Reply:
All good, I forgot about the sub-level error code.